### ISSUE TITLE
Brainstorm: (Token) economics

### ISSUE THREAD

**Issue by DecentralisedGaming** [DecentralisedGaming/Book#19]
**tl;dr**: we might need/ want 3 currencies

## Executive Summary
1. One coin for chain security.
2. One coin that scales with player base (and is somehow correlated to player base growth and player activity)
3. Perhaps also a stable coin. I suspect some players may prefer to hold (USD / EUR) rather than coin 1 (which is BTC or ETH like) or coin 2 (which inflates a lot). The stablecoin is the middle path. Coin 1. may rise in value over-time but it could also be highly volatile given what we see in current crypto markets. Coin 2 probably isn't so volatile and is the defacto norm for MMOs right now. I think this third option is a nice addition to MMOs.

## Background
These topics have come up before but I had some more crystallized thoughts recently. Let me brain dump some thoughts and then eventually tidy it up into a proper chapter.

For simplicity, I think we probably have to separate the in-game economics from that of the token economics. Not that they are independent, but rather the game economy itself can be complex but we need some way to deal with the token economics as an extra layer. We can gain insight from looking at existing online games and from looking at existing cryptocurrencies.

In addition, I think the economic concerns for a "NFT-game" will be slighty different than for a "fully decentralised game". I think we might need 3 currencies in a decentralised game. This is mainly due to the different purposes of each currency and the coupling between the underlying economies.

I'll split this into:

* Traditional games,
* NFT Games, and
* Dentralised games.

### Traditional games  
The traditional approach to MMO economies is to have an ever inflating supply of gold coins such that each coin is worth less in dollar terms each year. Obviously, this goes against what we see with crypto prices where people believe that the price of a crypto token should increase over time since they are scarce. Whenever I sold a rare gaming item to another player I was always interested in the USD price. When I was selling ores I didn't think about USD, but I think that mindset will change with decentralised games.

An inflating supply can also be an annoyance. In a game I played, I listed an item for 100,000 gp. That was a few years ago. The item didn't sell, but it is still listed. However, over that time the supply of gp has increased and is easier to obtain. Consequently, the item has been devalued since each gold piece is now worth less in dollar terms. So you really want to hold items in your inventory which don't suffer from inflation like gold pieces tend to do.

However, an inflating supply isn't totally insane. When new players join the game they want to start fighting monsters and collecting gp. They want to feel like progress is being made. If they rarely pick up gp then they will feel deflated (sorry, was an easy pun). I saw this in the same game I was just thinking on. During the testing phrase gp was hard to come by. Item pricing was all over the places and no one could really get a grasp on what prices should be. I think there is an argument for including an inflating currency, as we see in traditional MMOs, BUT players should also have a means of wealth protection at their disposal. Forcing players to hoard rares or "harder assets" is what MMOs implicitly do now but I'm not convinced that is a fair way to do it. This is an implied 'rule' that not everyone necessarily gets.

So how can things be different?

### NFT Game
These are blockchain games which are mostly centralised but have their assets represented by tokens then I think the in-game economics are basically the same as for a traditional online game. There is some concern required for the underlying chain, of course, but less so. If the price of Ethereum changes the the value of those NFT items could change but I don't think it is a guarantee by any means.

(I mostly skip analysis of these games for this book)

### Decentralised games  
For decentralised games I think the main economic concerns to address should be split into two factors:

1. the economics of the blockchain itself. Here the economics are about security.
2. the in-game economics, which is about the health of the player economy. 

I think these two economies should probably be decoupled from each other. In other words, I think the token for securing the chain should not be the main game currency or you need an extra mechanism (see below).

**Why?**
The currency for securing the chain should probably have low inflation. People should want to hold these tokens since they are scarce / rare and believe the tokens will go up in value (especially if the ecosystem grows and more players join). We see this most notably with Bitcoin but also with Ethereum. While the former has a pre-determined rate of issuance, the latter has a more ad-hoc rate that changes with "governance".

However, items in the game should be denominated in a different currency. Here is my thinking.
Just because the value of the underlying blockchain currency has appreciated it doesnt mean that the intrinsic utility of the item has improved. We've seen big jumps in crypto prices recently, so in gaming terms this could translate into items listed on an auction house becoming overvalued by (say) 50% over night in dollar terms. Utility didn't change, so it isn't like you can mine more rocks with the same pick axe. The ore that you collect doesn't give any better bonuses than it did before the underlying crypto token rocketed against the dollar. If you are a big holder of the underlying crypto token you won't care, but this could create a problem for new players.

But, there is a 'fix': players could just go the auction house and re-adjust all their prices! But that's a huge pain in the ass if it is manual. Given that blockchain games tend to be about relating in-game items to real values, then it could make sense to have auction houses denominated in USD / EUR etc. That wouldn't necessitate the introduction of a new currency, but it might be worth to include a stablecoin that's pegged to the USD as a means of exchange. Everyone knows the value of a dollar and it doesn't fluctuate like crypto does. I think it will be a headache to have an FX calculator open all the time if you price items in the network token and you just want to trade peer-to-peer (unless you include a USD price inside the trade window?)

So this potentially gives us two currencies to think about: the token for the network which is for security and performing actions, and a token that's pegged to fiat for in-game transactions. The first currency can be 'easily' printed by the network, while the second is not something that developers can really have much control over (although perhaps there is a monetisation angle in here for devs? that thought just occurred to me).

This leads me to think that there should be a third currency, which is basically a traditional MMO style currency which is inflationary by design. It will be easy to obtain and let's players get trading. Players wouldn't necessarily want to hold it long-term since the expectation is inflationary. It would be wiser to convert to the stabelcoin. The underlying network token may not necessarily increase in price either, so converting back player money back to that currency isn't necessarily wise. It isnt wise to pretend that players should hold that token as a means of protecting their wealth.

**Comment by DecentralisedGaming** [comment]
One interesting article that is similar in spirit is the following:

https://medium.com/moonchain-capital/how-enjin-is-helping-to-solve-inflation-in-virtual-economies-ebe6ae777462

The main idea for this piece is that ENJ tokens will somehow protect MMO economies. There is actually a lot of interesting pieces in the following article and some good points (i.e. a reserve currency which plays into some of what I say above). He digs into some of the history of the problem too, which I think is good. My contention is that ENJ by itself doesn't seem like the solution.

It sounds like ENJ would replace the network token in what I suggest above. It is some backing currency which is scarce. That isn't necessarily wrong, but I do suspect that you will need to augment it with an inflating currency. Also, ENJ is a lock-in of sorts as I understand it.
